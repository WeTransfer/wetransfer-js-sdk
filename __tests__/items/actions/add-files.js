const addFilesAction = require('../../../src/items/actions/add-files');
const { futureFile, RemoteFile } = require('../../../src/items/models');
const { RemoteTransfer } = require('../../../src/transfer/models');

describe('Add files action', () => {
  let addFiles = null;
  let sendItems = null;
  let transfer = null;
  beforeEach(() => {
    sendItems = jest.fn().mockReturnValue([
      {
        id: 'random-hash',
        content_identifier: 'file',
        local_identifier: 'delightful-cat',
        meta: {
          multipart_parts: 3,
          multipart_upload_id: 'some.random-id--'
        },
        name: 'kittie.gif',
        size: 195906,
        upload_id: 'more.random-ids--',
        upload_expires_at: 1520410633
      }
    ]);

    addFiles = addFilesAction({
      sendItems,
      futureFile,
      RemoteFile
    });

    transfer = new RemoteTransfer();
  });

  it('should create an add items request', async () => {
    const linkFiles = await addFiles(transfer, [
      {
        url: 'https://wetransfer.com',
        meta: {
          title: 'WeTransfer'
        }
      }
    ]);
    expect(linkFiles).toMatchSnapshot();
  });
});
