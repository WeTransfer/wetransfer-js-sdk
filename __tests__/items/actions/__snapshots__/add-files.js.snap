// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add files action should create an add items request 1`] = `
Array [
  Object {
    "content_identifier": "file",
    "id": "random-hash",
    "local_identifier": "delightful-cat",
    "meta": Object {
      "multipart_parts": 3,
      "multipart_upload_id": "some.random-id--",
    },
    "name": "kittie.gif",
    "size": 195906,
  },
]
`;
